### PR TITLE
Fix build system

### PR DESCRIPTION
**Pull request by dacav**
In this PR I'm addressing some little problems I've found when bootstrapping the library.
This is effectively fixing https://github.com/bakulf/libnxml/issues/2

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,68 @@
+Makefile.in
+aclocal.m4
+autom4te.cache/
+compile
+config.guess
+config.h.in
+config.sub
+configure
+depcomp
+doc/Makefile.in
```
Comment (bakulf) [inline .gitignore]: Just `Makefile.in`. Remove this and any other occurrencies of Makefile.in in this file.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+install-sh
+ltmain.sh
+m4/
+missing
+src/Makefile.in
```
Comment (bakulf) [inline .gitignore]: remove this and the following one.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+test/Makefile.in
+Makefile
```
Comment (bakulf) [inline .gitignore]: Same here: Just `Makefile` and remove any other ones.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+config.h
+config.log
+config.status
+doc/Makefile
+doxy.conf
+libtool
+nxml.pc
+src/.deps/
```
Comment (bakulf) [inline .gitignore]: .deps
.libs

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+src/.libs/
+src/Makefile
+src/libnxml.la
+src/nxml.h
+src/nxml_download.lo
```
Comment (bakulf) [inline .gitignore]: *.lo
*.o

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+src/nxml_string.lo
+src/nxml_string.o
+src/nxml_tools.lo
+src/nxml_tools.o
+src/nxml_utf.lo
+src/nxml_utf.o
+src/nxml_write.lo
+src/nxml_write.o
+stamp-h1
+
```
Comment (bakulf) [inline .gitignore]: No empty line.

**Review by bakulf — CHANGES_REQUESTED** [review]
Thanks for doing this. A couple of changes in .gitignore

**Comment by dacav** [comment]
All solved.  I'm not sure how your merge policy is, but I recommend a `rebase --autosquash` :)

Cheers

**Comment by dacav** [comment]
I've added a bunch of further fixes...
I had some problems in adding the unit tests that I'm writing to tap the crashes.

**Comment by dacav** [comment]
The recently pushed clang-format is breaking this PR horribly, so I'm closing it in favour of a new one.

**Pull request closed without merging**